### PR TITLE
add expresso 3.1

### PR DESCRIPTION
**Pull request by vmvmpvm**
Adding http://www.ultrapico.com/expresso.htm

From their home page:
Expresso editor is equally suitable as a teaching tool for the beginning user of regular expressions or as a full-featured development environment for the experienced programmer or web designer with an extensive knowledge of regular expressions.

The license refers to Expresso 3.0, but that is the only license I could find.

### CODE REVIEW

File: expresso.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+    "homepage": "http://www.ultrapico.com/Expresso.htm",
+    "description": "Expresso editor is equally suitable as a teaching tool for the beginning user of regular expressions or as a full-featured development environment for the experienced programmer or web designer with an extensive knowledge of regular expressions.",
```
Comment (Ash258) [inline expresso.json]:
```suggestion
    "description": "Editor equally suitable as a teaching tool for the beginning user of regular expressions or as a full-featured development environment for the experienced programmer or web designer with an extensive knowledge of regular expressions.",
```

File: expresso.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    "bin": "expresso.exe",
+    "shortcuts": [
+        [
+            "expresso.exe",
+            "Expresso"
+        ]
+    ],
+    "checkver": {
+        "url": "http://www.ultrapico.com/ExpressoDownload.htm",
+        "re": "Version ([\\d\\.]+)"
```
Comment (Ash258) [inline expresso.json]:
```suggestion
        "regex": "Version ([\\d.]+)"
```

**Comment by vmvmpvm** [comment]
> Free registration is not optional:
> 
> > If you have not registered, Version 3.1.6224 will expire after 60 days or on January 1, 2020, whichever comes first

They also state below in their registration page, not sure what they actually mean, I had my copy registered so I do not know how it behaves after expiry. And so at this point I am not sure what the note in the manifest should say.

> Registration of Expresso is optional and free of charge. After a trial period you will receive reminders > to register your copy.

File: bucket/expresso.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
+    "homepage": "http://www.ultrapico.com/Expresso.htm",
+    "description": "Editor equally suitable as a teaching tool for the beginning user of regular expressions or as a full-featured development environment for the experienced programmer or web designer with an extensive knowledge of regular expressions.",
+    "version": "3.1.6224",
+    "notes": "Free registration is optional, can be done at http://www.ultrapico.com/App/Register.aspx",
```
Comment (Ash258) [inline bucket/expresso.json]:
```suggestion
```

**Pull request merged**